Java

```diff
@@ -47,34 +58,27 @@ class student{
 		
 		
 		
-		
       	for(int t=0; t<w; t++) {
       	
           
         	System.out.print("Attendance for week # "+(t+1)+": ");
-        	time[t] = s.nextDouble(); 
-        
+        	double hours = s.nextDouble(); 
+        	students[i].addHours(t,hours);
+      	}
+	
         	
-      }
+  	
+        for(int t=0; t<n; t++){
 
-
-        
-        for(int t=0; t<w; t++){
-
-          total = total + time[t];
-
-        }          
-        double average = total / w;
-        double perc = average*10;
-       
+       students[i].addHours(double totalHours);
```

Comment:
A similar loop over the weeks is done above at line 61: this should be removed (let's not forget the closing brace).